Tests/Util/FilenameUtilsTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Vich\UploaderBundle\Tests\Util;
+
+use PHPUnit\Framework\TestCase;
+use Vich\UploaderBundle\Util\FilenameUtils;
+
+class FilenameUtilsTest extends TestCase
+{
+    /**
+     * @dataProvider spitNameByExtensionProvider
+     */
+    public function spitNameByExtension($filename, $basename, $extension)
+    {
+        $this->assertSame([$basename, $extension], FilenameUtils::spitNameByExtension($filename));
+    }
+
+    public function spitNameByExtensionProvider()
+    {
+        return [
+            ['Текстовый файл.txt', 'Текстовый файл', 'txt'],
```

Comment:
But it does not explain the cases you're providing to it. You should imagine that instead of that single test you get 5 tests (one for each row provided by the data provider). In such situation, the 5 tests would get a different name explaining the single case you're testing (and maybe what you expect). Since there is only one test, we need to move such help to the data provider, otherwise a possible failing test would tell not enough.
See example 2.6 on https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers